src/KubernetesClient/Informers/ResourceObservableExtensions.cs

```diff
@@ -0,0 +1,294 @@
+                MaxDegreeOfParallelism = parallelWorkers
+            });
+            actionBlock.Completion.ContinueWith(x =>
+            {
+                if (x.IsFaulted) logger.LogCritical(x.Exception.Flatten(), "Controller encountered a critical error");
+            });
+            return workerQueue.LinkTo(actionBlock, new DataflowLinkOptions {PropagateCompletion = true});
+        }
+
+        public static IObservable<ResourceEvent<TResource>> WithResets<TResource>(this IObservable<ResourceEvent<TResource>> source, Func<List<ResourceEvent<TResource>>, IEnumerable<ResourceEvent<TResource>>> action)
+        {
+            return source.WithResets(action, e => e);
+        }
+
+        public static IObservable<TOut> WithResets<TResource, TOut>(this IObservable<ResourceEvent<TResource>> source, Func<List<ResourceEvent<TResource>>, IEnumerable<ResourceEvent<TResource>>> resetSelector, Func<ResourceEvent<TResource>, TOut> itemSelector)
+        {
+            var resetBuffer = new List<ResourceEvent<TResource>>();
+
+            return Observable.Create<TOut>(observer =>
+                {
+                    void FlushBuffer()
+                    {
+                        if (!resetBuffer.Any())
+                            return;
+
+                        foreach (var item in resetSelector(resetBuffer))
+                            observer.OnNext(itemSelector(item));
+                        resetBuffer.Clear();
+                    }
+
+                    void OnComplete()
+                    {
+                        FlushBuffer();
+                        observer.OnCompleted();
+                    }
+
+                    void OnError(Exception e)
+                    {
+                        observer.OnError(e);
+                    }
+
+                    var upstreamSubscription = source
+                        .Subscribe(notification =>
+                        {
+                            if (notification.EventFlags.HasFlag(EventTypeFlags.Reset))
+                            {
+                                resetBuffer.Add(notification);
+                                if (!notification.EventFlags.HasFlag(EventTypeFlags.ResetEnd)) // continue buffering till we reach the end of list window
+                                    return;
+                            }
+
+                            if (notification.EventFlags.HasFlag(EventTypeFlags.ResetEnd))
+                            {
+                                FlushBuffer();
+                                return;
+                            }
+
+                            if (!notification.EventFlags.HasFlag(EventTypeFlags.Reset) && resetBuffer.Count > 0) FlushBuffer();
+
+                            observer.OnNext(itemSelector(notification));
+                        }, OnError, OnComplete);
+                    return new CompositeDisposable(upstreamSubscription, Disposable.Create(OnComplete));
```

Comment:
Consider `StableCompositeDisposable.Create` instead to create disposables from a fixed number of disposables.